File: clojure/src/spoonacular_api/specs/parse_ingredients_200_response_inner.clj
(ns spoonacular-api.specs.parse-ingredients-200-response-inner
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.parse-ingredients-200-response-inner-estimated-cost :refer :all]
            [spoonacular-api.specs.parse-ingredients-200-response-inner-nutrition :refer :all]
            )
  (:import (java.io File)))


(def parse-ingredients-200-response-inner-data
  {
   (ds/req :id) int?
   (ds/req :original) string?
   (ds/req :originalName) string?
   (ds/req :name) string?
   (ds/req :nameClean) string?
   (ds/req :amount) float?
   (ds/req :unit) string?
   (ds/req :unitShort) string?
   (ds/req :unitLong) string?
   (ds/req :possibleUnits) (s/coll-of string?)
   (ds/req :estimatedCost) parse-ingredients-200-response-inner-estimated-cost-spec
   (ds/req :consistency) string?
   (ds/req :aisle) string?
   (ds/req :image) string?
   (ds/req :meta) (s/coll-of string?)
   (ds/req :nutrition) parse-ingredients-200-response-inner-nutrition-spec
   })

(def parse-ingredients-200-response-inner-spec
  (ds/spec
    {:name ::parse-ingredients-200-response-inner
     :spec parse-ingredients-200-response-inner-data}))

File: scala/src/main/scala/com/spoonacular/client/model/ParseIngredients200ResponseInner.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import com.spoonacular.client.model.BigDecimal
import com.spoonacular.client.model.ParseIngredients200ResponseInnerEstimatedCost
import com.spoonacular.client.model.ParseIngredients200ResponseInnerNutrition
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param original 
 * @param originalName 
 * @param name 
 * @param nameClean 
 * @param amount 
 * @param unit 
 * @param unitShort 
 * @param unitLong 
 * @param possibleUnits 
 * @param estimatedCost 
 * @param consistency 
 * @param aisle 
 * @param image 
 * @param meta 
 * @param nutrition 
 */
case class ParseIngredients200ResponseInner(id: Int,
                original: String,
                originalName: String,
                name: String,
                nameClean: String,
                amount: BigDecimal,
                unit: String,
                unitShort: String,
                unitLong: String,
                possibleUnits: Seq[String],
                estimatedCost: ParseIngredients200ResponseInnerEstimatedCost,
                consistency: String,
                aisle: String,
                image: String,
                meta: Seq[String],
                nutrition: ParseIngredients200ResponseInnerNutrition
                )

object ParseIngredients200ResponseInner {
    /**
     * Creates the codec for converting ParseIngredients200ResponseInner from and to JSON.
     */
    implicit val decoder: Decoder[ParseIngredients200ResponseInner] = deriveDecoder
    implicit val encoder: ObjectEncoder[ParseIngredients200ResponseInner] = deriveEncoder
}
